src/Honeycomb.Samplers/DeterministicSampler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using OpenTelemetry.Trace;
+
+namespace Honeycomb.Samplers
+{
+    public class DeterministicSampler : Sampler, IDisposable
+    {
+        private const string DescriptionFormat = "DeterministicSampler({0})";
+        private const string SampleRateAttributeName = "sampleRate";
+        private const string Hphen = "-";
+        private const int Zero = 0;
+        private const int Four = 4;
+        private const int One = 1;
+        private const int Base16 = 16;
+        private readonly SHA1 sha1 = SHA1.Create();
+        private readonly int sampleRate;
+        private readonly long upperBound;
+
+        public DeterministicSampler(int sampleRate)
+        {
+            if (sampleRate < Zero)
+            {
+                throw new ArgumentOutOfRangeException("Sample rate must not be negative.");
+            }
+
+            this.sampleRate = sampleRate;
+            this.Description = string.Format(DescriptionFormat, this.sampleRate.ToString(CultureInfo.InvariantCulture));
+            this.upperBound = sampleRate == Zero ? Zero : uint.MaxValue / sampleRate;
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            if (sampleRate == One)
+            {
+                return CreateResult(SamplingDecision.RecordAndSample, One);
+            }
+            if (sampleRate == One)

```

Comment:
Not sure this is right... you have sample rate checking twice, the second won't be hit.

Reply:
Should that be `sampleRate == Zero`?